packages/db/tests/models/test-granule-model.js

```diff
@@ -49,21 +57,251 @@ test.after.always(async (t) => {
     granulePgModel,
     collectionCumulusId,
     executionCumulusId,
+    granuleExecutionHistoryPgModel,
   } = t.context;
 
   const granule = fakeGranuleRecordFactory({
     collection_cumulus_id: collectionCumulusId,
+    status: 'running',
+  });
+
+  const [granuleCumulusId] = await granulePgModel.create(
+    knex,
+    granule
+  );
+
+  await granuleExecutionHistoryPgModel.create(knex, {
+    granule_cumulus_id: granuleCumulusId,
     execution_cumulus_id: executionCumulusId,
+  });
+
+  const [granuleWithHistory] = await granulePgModel.getWithExecutionHistory(
+    knex,
+    granule
+  );
+
+  t.like(
+    granuleWithHistory,
+    {
+      ...granule,
+      cumulus_id: granuleCumulusId,
+      execution_cumulus_ids: [executionCumulusId],
+    }
+  );
+});
+
+test('getWithExecutionHistory returns execution history for granule with multiple executions', async (t) => {
+  const {
+    knex,
+    granulePgModel,
+    collectionCumulusId,
+    executionCumulusId,
+    executionPgModel,
+    granuleExecutionHistoryPgModel,
+  } = t.context;
+
+  const granule = fakeGranuleRecordFactory({
+    collection_cumulus_id: collectionCumulusId,
     status: 'running',
   });
 
-  await granulePgModel.upsert(knex, granule);
+  const [granuleCumulusId] = await granulePgModel.create(
+    knex,
+    granule
+  );
+
+  await granuleExecutionHistoryPgModel.create(knex, {
+    granule_cumulus_id: granuleCumulusId,
+    execution_cumulus_id: executionCumulusId,
+  });
+
+  const [newExecutionCumulusId] = await executionPgModel.create(
+    knex,
+    fakeExecutionRecordFactory()
+  );
+
+  await granuleExecutionHistoryPgModel.create(knex, {
+    granule_cumulus_id: granuleCumulusId,
+    execution_cumulus_id: newExecutionCumulusId,
+  });
+
+  const [granuleWithHistory] = await granulePgModel.getWithExecutionHistory(
+    knex,
+    granule
+  );
+
+  t.like(
+    granuleWithHistory,
+    {
+      ...granule,
+      cumulus_id: granuleCumulusId,
+      execution_cumulus_ids: [executionCumulusId, newExecutionCumulusId],
+    }
+  );
+});
+
+test('GranulePgModel.createWithExecutionHistory() creates a new granule with execution history', async (t) => {
+  const {
+    knex,
+    granulePgModel,
+    collectionCumulusId,
+    executionCumulusId,
+    granuleExecutionHistoryPgModel,
+  } = t.context;
+
+  const granule = fakeGranuleRecordFactory({
+    collection_cumulus_id: collectionCumulusId,
+    status: 'running',
+  });
+
+  const [granuleCumulusId] = await granulePgModel.createWithExecutionHistory(
+    knex,
+    granule,
+    executionCumulusId
+  );
+
+  const granuleRecord = await granulePgModel.get(
+    knex,
+    granule
+  );
+
+  t.like(
+    granuleRecord,
+    {
+      ...granule,
+      cumulus_id: granuleCumulusId,
+    }
+  );
+  t.deepEqual(
+    await granuleExecutionHistoryPgModel.search(
+      knex,
+      { granule_cumulus_id: granuleCumulusId }
+    ),
+    [{
+      granule_cumulus_id: Number.parseInt(granuleCumulusId, 10),
```

Comment:
Curious - why `parseInt` over Number(granuleCumlusId)  here?    Normally I'd grok you want true to result in NaN instead of 1 or something, but there's the case of strings like '32xfoo' that go the other way, and we should be fairly certain this is a generated numeric ID string.  

No requested change specifically.

Reply:
The whole reason for doing this at all is that the `cumulus_id` as a bigInteger value is actually returned by Knex as a string, not a number. It is of course actually stored by the database as a number, so it's a quirk of Knex not the database. Thus when we want to write a FK (or compare one) pointing to the granules table, we have to make sure to convert the string back to a number.

Does that answer the question?

Reply:
Right, I get that knex is returning it as a string, I was just curious if there was an explicit reason for 

` Number.parseInt(granuleCumulusId, 10)` 

over the more terse/handling mixed strings differently

`Number(granuleCumulusId)` 

in this instance, given we know knex is returning a stringified bigInt.      

Again, not a requested change or necessarily a nit.

Reply:
I honestly didn't know that just doing `Number()` would even work. But I'm happy to change it

Reply:
No worries, this is just a curiosity/code pattern convo, we can have it elsewhere as it's not directly relevant to this PR getting merged.